tests/mitol/payment_gateway/test_cybersource.py

```diff
@@ -212,3 +220,133 @@ def test_cybersource_payment_response(order, cartitems):
     )
 
     assert response.state == ProcessorResponse.STATE_CANCELLED
+
+
+def test_cybersource_client_configuration_url(settings):
+    """
+    Tests that get_client_configuration generates the right configuration url for CyberSource Payment processor
+    """
+    settings.CYBERSOURCE_REST_API_ENVIRONMENT = "apitest.cybersouece.com"
+    gateway = CyberSourcePaymentGateway()
+    configuration = gateway.get_client_configuration()
+    assert (
+        configuration["run_environment"]
+        == settings.MITOL_PAYMENT_GATEWAY_CYBERSOURCE_REST_API_ENVIRONMENT
+    )
+
+
+def test_generate_refund_payload(refund):
+    """
+    Tests that generate_refund_payload function generates correct readable formatted JSON for the request to
+    CyberSource refund API calls.
+    """
+
+    cybersource_gateway = CyberSourcePaymentGateway()
+    refund_payload = json.loads(cybersource_gateway.generate_refund_payload(refund))
+    assert "_client_reference_information" in refund_payload
+    assert "_order_information" in refund_payload
+
+    client_reference_information = refund_payload["_client_reference_information"]
+
+    # Passing transaction_id in reference information is important because the refund API won't send a duplicate
+    # request response if there is no transaction_id in the request payload
+    assert client_reference_information["_transaction_id"] == refund.transaction_id
+
+    order_information = refund_payload["_order_information"]
+    assert "_amount_details" in order_information
+    assert order_information["_amount_details"] == {
+        "_total_amount": refund.refund_amount,
+        "_currency": refund.refund_currency,
+    }
+
+
+def test_cybersource_refund_response_success(refund, mocker):
+    """
+    Tests that the perform_refund method returns the expected response to the calling application.
+    This test uses a sample request JSON to as API response from CyberSource since we won't be hitting the
+    real API in tests. Instead, We will mock it for this case
+    """
+    payload_rf = open(
+        os.getcwd() + "/tests/mitol/payment_gateway/test_refund_success.json", mode="r"
+    )
+    test_refund_success = json.load(payload_rf)
```

Comment:
I'd suggest refactoring this setup code into a fixture since you have the same pattern in several test functions, you can then use pytest's [indirection parametrization](https://docs.pytest.org/en/latest/example/parametrize.html#apply-indirect-on-particular-arguments) on each test function, so it would look something like this:


```python
@pytest.fixture
def response_payload(request):
    # use open() as a context manager here so it gets cleaned up, your original code left the file open
    with open(os.path.join(
        os.getcwd(),
        "tests/mitol/payment_gateway",
        f"{request.param}.json"
    ), mode="r") as f:
        response_txt = f.read()
        response_json = json.loads(response_txt)
        yield (response_txt, response_json)


@pytest.mark.parametrize("response_payload", ["test_refund_success"], indirect=True)
def test_cybersource_refund_response_success(response_payload, refund, mocker):
    response_txt, response_json = response_payload
    
    assert "orderInformation" in response_json
    assert "refundAmountDetails" in response_json
    assert "status" in response_json

    mocker.patch(
        "CyberSource.RefundApi.refund_payment",
        return_value=["", "", response_txt],
    )
    cybersource_gateway = CyberSourcePaymentGateway()
    response = cybersource_gateway.perform_refund(refund)

    assert response.state == ProcessorResponse.STATE_PENDING
    assert response.transaction_id == refund.transaction_id
```

Reply:
I've used the indirect approach for the fixtures. Ther were some old payment tests that were reading files so I've refactored them too to use this.